### PR TITLE
Add logger indent option and indent nested xcode command outputs

### PR DESCRIPTION
**Pull request by xster**
#7860

Putting all the flutter outputs and nested command outputs all together is too confusing. Indent nested messages.

Looks like:
```
...
Launching lib/main.dart on iPhone in debug mode...
building app com.yourcompany.helloWorld, mode BuildMode.debug, target /Users/xster/Development/flutter/examples/hello_world/lib/main.dart
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


    The following build commands failed:
    	Check dependencies
    (1 failure)
Xcode's output:
↳
    Build settings from command line:
        ARCHS = arm64
        ONLY_ACTIVE_ARCH = YES
        SDKROOT = iphoneos10.2

    === CLEAN TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===

    Check dependencies
    [BCEROR]Signing for "Runner" requires a development team. Select a development team in the project editor.
    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.2'
    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.2'
    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.2'
...
```

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 1]
```diff
@@ -28,7 +28,9 @@ abstract class Logger {
 
   /// Display normal output of the command. This should be used for things like
   /// progress messages, success messages, or just normal command output.
-  void printStatus(String message, { bool emphasis: false, bool newline: true, String ansiAlternative });
+  void printStatus(
+    String message,
+    { bool emphasis: false, bool newline: true, String ansiAlternative, int indent });
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: nit: newline before )

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 2]
```diff
@@ -65,13 +67,17 @@ class StdoutLogger extends Logger {
   }
 
   @override
-  void printStatus(String message, { bool emphasis: false, bool newline: true, String ansiAlternative }) {
+  void printStatus(
+    String message,
+    { bool emphasis: false, bool newline: true, String ansiAlternative, int indent }) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: ditto

the rule is just to make sure the whitespace at the open bracket matches the whitespace at the close bracket.

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 3]
```diff
@@ -113,7 +119,9 @@ class BufferLogger extends Logger {
   void printError(String message, [StackTrace stackTrace]) => _error.writeln(message);
 
   @override
-  void printStatus(String message, { bool emphasis: false, bool newline: true, String ansiAlternative }) {
+  void printStatus(
+    String message,
+    { bool emphasis: false, bool newline: true, String ansiAlternative, int indent }) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: ditto here and below

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 4]
```diff
@@ -165,10 +165,13 @@ Future<XcodeBuildResult> buildXcodeProject({
   );
 
   if (result.exitCode != 0) {
+    printStatus('Failed to build iOS app');
     if (result.stderr.isNotEmpty)
-      printStatus(result.stderr);
+      printStatus('Error output from Xcode build:\n↳');
+      printStatus(result.stderr, indent: 4);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/mac.dart]: i'm guessing you're missing some braces here

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 1 above)
Comment (xster) [inline packages/flutter_tools/lib/src/base/logger.dart]: Understood

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 2 above)
Comment (xster) [inline packages/flutter_tools/lib/src/base/logger.dart]: Done

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 3 above)
Comment (xster) [inline packages/flutter_tools/lib/src/base/logger.dart]: Done

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 4 above)
Comment (xster) [inline packages/flutter_tools/lib/src/ios/mac.dart]: Doh! Thanks

**Comment by Hixie** [comment]
LGTM

**Pull request merged**